config/routes.rb

```diff
@@ -1,3 +1,7 @@
 Rails.application.routes.draw do
+  root 'homes#index'
+  devise_for :users
```

Comment:
- [ ] A redundant message `You need to sign in or sign up before continuing.` is showing once I landed the sign up page or when I sign out. This message should only show when the user tries to go a page that he/she can't see. This behavior because we have default root which to posts which redirect us back to sign_in page. We can get rid of this message by using `authenticated` devise helper in routes, that way we have different roots for authenticated and unauthenticated users.
```
devise_scope :user do
  root to: 'devise/sessions#new'
end
authenticated :user do
  root 'homes#index'
end
```